src/test/java/com/google/sps/data/BooksManagerTest.java

```diff
@@ -0,0 +1,88 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BooksManagerTest {
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private final BooksManager booksManager = new BooksManager();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testGetLibrariesForBook() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity bookEntity = new Entity("Book");
+    bookEntity.setProperty("bookId", "1234");
+    datastore.put(bookEntity);
+    Entity library1 = new Entity("LibraryStock", bookEntity.getKey());
+    library1.setProperty("stock", 3);
+    datastore.put(library1);
+    Entity library2 = new Entity("LibraryStock", bookEntity.getKey());
+    library2.setProperty("stock", 0);
+    datastore.put(library2);
+    Entity library3 = new Entity("LibraryStock", bookEntity.getKey());
+    library3.setProperty("stock", 7);
+    datastore.put(library3);
+    Assert.assertEquals(Arrays.asList(library1, library3), booksManager.getLibrariesForBook("1234"));
+  }
+
+  @Test
+  public void testRemoveBookFromLibrary() throws BadRequestException {
```

Comment:
Maybe also verify that this doesn't remove stock for a book in a different library than the one you requested it from? (could be a separate test, not sure what makes sense).

And also a test on what happens if you try to remove a book from a library which does not have it.

Reply:
Added those 2 tests.